File: trunk/redemption/backend/email_service.php
<?php
	require_once 'settings.php';
	
	
	class EmailServiceAttachmentParams
	{
		public $rid;
		public $sid;
		public $fileName;
		
		public function __construct($rid = NULL, $fileName = NULL, $sid = NULL)
		{
			$this->rid = $rid;
			$this->sid = $sid;
			$this->fileName = $fileName;
		}
	}
	
	class EmailServiceParams
	{
		public $from = NULL;
		public $to = array();
		public $subject = '';
		public $messageHTML = '';
		public $attachments = array();
	}
	
	class EmailServiceResult
	{
		public $errorCode;
		public $errorMessage;
	}
	
	class EmailService
	{
		/**
		 * @param EmailServiceParams $params
		 * @return EmailServiceResult
		 */
		public static function sendMail($params)
		{
			return json_decode(file_get_contents(
					Settings::SERVICE_SEND_MAIL .
					'?params=' . urlencode(json_encode($params))
			));
		}
	}
?>

File: trunk/GenesysService/src/backend/Email_Service.java
package backend;

import java.io.InputStreamReader;
import java.net.URL;

import backend.Settings.Result;

import com.google.gson.Gson;


public class Email_Service {
	class EmailServiceAttachmentParams
	{
		public rid;
		public sid;
		public String fileName;
		
		public EmailServiceAttachmentParams(rid,String fileName, sid)
		{
			this.rid = rid;
			this.sid = sid;
			this.fileName = fileName;
		}
	}
	
	class EmailServiceParams
	{
		public String from = null;
		public String[] to;
		public String subject = "";
		public String messageHTML = "";
		public attachments = array();
	}
	
	class EmailServiceResult
	{
		public String errorCode;
		public String errorMessage;
	}
	
	public class EmailService
	{
		/**
		 * @param EmailServiceParams params
		 * @return EmailServiceResult
		 */
		public EmailServiceResult sendMail(EmailServiceParams params)
		{
			Gson gson = new Gson();
			URL search = new URL(Settings.SERVICE_SEND_MAIL + "?params=" + gson.toJson(params));
			InputStreamReader output = new InputStreamReader(search.openStream());
			return gson.fromJson(output, EmailServiceResult.class);
		}
	}
}
}
